File: flapypan-blog-api/src/main/java/top/flapypan/blog/entity/Setting.java
package top.flapypan.blog.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.*;

import java.util.Objects;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "t_setting")
public class Setting {

    @Id
    @Column(name = "s_key")
    private String key;

    @Column(name = "s_value")
    private String value;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Setting setting = (Setting) o;
        return Objects.equals(key, setting.key);
    }

    @Override
    public int hashCode() {
        return key.hashCode();
    }
}

File: flapypan-blog-api/src/main/kotlin/top/flapypan/blog/entity/Setting.kt
package top.flapypan.blog.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

/**
 * 设置实体类
 */
@Entity
@Table(name = "t_setting")
class Setting {

    @Id
    @Column(name = "s_key")
    var key: String = ""

    @Column(name = "s_value")
    var value: String? = null

    constructor()

    constructor(key: String, value: String?) {
        this.key = key
        this.value = value
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Setting
        return key == other.key
    }

    override fun hashCode() = key.hashCode()

}
